web/src/services/Selector.service.ts

```diff
@@ -0,0 +1,87 @@
+import {isNumber} from 'lodash';
```

Comment:
Separating some of the logic from the Assertion service as it makes more sense to have one for selectors